.github/workflows/autoMerge.yml

```diff
@@ -8,9 +12,23 @@ permissions:
 jobs:
   automerge:
     runs-on: ubuntu-latest
-    if: github.event.pull_request.user.login == 'github-actions[bot]'
+
     steps:
-      - uses: peter-evans/enable-pull-request-automerge@v3
+      - name: Get Pull Request Commits
+        id: get_pr_commits
+        run: |
+          PR_COMMITS=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            "https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/commits")
+          echo "::set-output name=commit_author::$(echo "$PR_COMMITS" | jq -r '.[0].commit.author.name')"
+        
+      - name: Print Commit Author
+        run: |
+          echo "Commit Author Value: ${{ steps.get_pr_commits.outputs.commit_author }}" 
+        
+      - name: Merge Pull Request
+        if: steps.get_pr_commits.outputs.commit_author == 'github-actions[bot]'
```

Comment:
Is this secure? Anybody can open a PR with a commit having this particular author, and then that PR would be auto-merged.